### PR TITLE
Match `grtluigi`

### LINKED ISSUE #598: Match gr/grtluigi.c
- [x] [`func_80221C10`](https://decomp.me/scratch/L9klz)
- [x] [`lbl_80221C14`](https://decomp.me/scratch/vGpE6)
- [x] [`lbl_80221C84`](https://decomp.me/scratch/F2sax)
- [x] [`lbl_80221C88`](https://decomp.me/scratch/74RUK)
- [x] [`lbl_80221CAC`](https://decomp.me/scratch/4yhKI)
- [x] [`func_80221CB4`](https://decomp.me/scratch/wXG0o)
- [x] [`lbl_80221D9C`](https://decomp.me/scratch/ai4GG)
- [x] [`lbl_80221DC8`](https://decomp.me/scratch/Glhco)
- [x] [`lbl_80221DD0`](https://decomp.me/scratch/Tv62A)
- [x] [`lbl_80221DD4`](https://decomp.me/scratch/8p8gC)
- [x] [`lbl_80221DD8`](https://decomp.me/scratch/zJQC5)
- [x] [`lbl_80221E28`](https://decomp.me/scratch/NKF9a)
- [x] [`lbl_80221E30`](https://decomp.me/scratch/3UciJ)
- [x] [`lbl_80221E64`](https://decomp.me/scratch/GgNzz)
- [x] [`lbl_80221E68`](https://decomp.me/scratch/6d9Di)
- [x] [`lbl_80221EB8`](https://decomp.me/scratch/rEeRe)
- [x] [`lbl_80221EC0`](https://decomp.me/scratch/AeHLp)
- [x] [`lbl_80221EE0`](https://decomp.me/scratch/f3uZM)
- [x] [`lbl_80221EE4`](https://decomp.me/scratch/yeq3w)
- [x] [`lbl_80221EEC`](https://decomp.me/scratch/11yWE)

### PR DESCRIPTION
**Pull request by ribbanya**
Depends on #699. Resolves #598.

```yml
Co-authored-by: jeffm <jeffmao99@berkeley.edu>
```

### CODE REVIEW

File: src/melee/gr/grtluigi.c [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+void lbl_80221E68(HSD_GObj* gobj)
+{
+    /// @todo Unused stack.
+#ifdef MUST_MATCH
+    u8 unused[8];
+#endif
+
+    Map* map = gobj->user_data;
+    func_801C2ED0(gobj->hsd_obj, map->map_id);
+    func_801C8138(gobj, map->map_id, 0);
```
Comment (r-burns) [inline src/melee/gr/grtluigi.c:155]: We can probably fix the stack here using our new gobj getters + C casts

**Review by r-burns — APPROVED** [review]

File: src/melee/gr/grtluigi.c (hunk 1 above)
Comment (ribbanya) [reply src/melee/gr/grtluigi.c:155]: That's its own inline, appear a lot in these target stages.

Comment (ribbanya) [reply src/melee/gr/grtluigi.c]: For some reason there's still 4 bytes of unused stack in 8d3fc0408ed0264d00ba599fec79e0b3cdb4fad3.

Comment (r-burns) [reply src/melee/gr/grtluigi.c]: Yeah I'm thinking you have to use the getter for hsd_obj as well.

File: src/melee/gr/grtluigi.c [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+void lbl_80221DD8(HSD_GObj* gobj)
+{
+    /// @todo Unused stack.
+#ifdef MUST_MATCH
+    u8 unused[8];
+#endif
+
+    Map* map = gobj->user_data;
+    func_801C2ED0(gobj->hsd_obj, map->map_id);
+    func_801C8138(gobj, map->map_id, 0);
```
Comment (r-burns) [inline src/melee/gr/grtluigi.c:131]: Can probably use the inline here too

**Review by r-burns — APPROVED** [review]

**Pull request merged**